content/src/debug.js

```diff
@@ -123,15 +134,17 @@ var debug = window.ide = {
       panel: embedded ? 'auto' : true
     };
   },
-  trace: function(event, data) {
+  trace: function(event, data) { 
     detectStuckProgram();
     // This receives events for the new debugger to use.
     currentDebugId += 1;
-    var record = {line: 0, eventIndex: null, startCoords: [], endCoords: [], method: "", data: "", seeeval:false};
+    var record = {line: 0, eventIndex: null, startCoords: [], endCoords: [], method: "", 
+        data: "", seeeval:false};
     traceEvents.push(event);
     currentEventIndex = traceEvents.length - 1;
     record.eventIndex = currentEventIndex;
     var lineno = traceEvents[currentEventIndex].location.first_line;
+    setTimeout(function() {view.createSlider(traceEvents, arrows, view.paneid("left"), debugRecordsByLineNo, targetWindow)}, 1000);
```

Comment:
Make a function to manage this timer: ideally there should be only one timeout active at once.  You can cancel the old timeout if you come across this trace function again.

Something like `sliderTimer = null; function setupSlider() { if (silderTimer) 
{ clearTimeout(sliderTimer); sliderTimer = null; } sliderTimer = setTimeout(....) };`